packages/plugin-ext/src/main/browser/custom-editors/custom-editor-widget.ts

```diff
@@ -0,0 +1,116 @@
+/********************************************************************************
+ * Copyright (c) 2021 SAP SE or an SAP affiliate company and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject, postConstruct } from 'inversify';
+import URI from '@theia/core/lib/common/uri';
+import { CopyFileOptions, FileOperation } from '@theia/filesystem/lib/common/files';
+import { NavigatableWidget, Saveable, SaveableSource } from '@theia/core/lib/browser';
+import { CustomEditorService } from '../custom-editors/custom-editor-service';
+import { Reference } from '@theia/core/lib/common/reference';
+import { WebviewWidget } from '../webview/webview';
+import { UndoRedoService } from './undo-redo-service';
+import { CustomEditorModel } from './custom-editors-main';
+
+@injectable()
+export class CustomEditorWidget extends WebviewWidget implements SaveableSource, NavigatableWidget {
+    id: string;
+    static FACTORY_ID = 'plugin-custom-editor';
+
+    @inject(CustomEditorService)
+    protected readonly customEditorService: CustomEditorService;
+
+    @inject(UndoRedoService)
+    protected readonly undoRedoService: UndoRedoService;
+
+    resource: URI;
+
+    protected _modelRef: Reference<CustomEditorModel>;
+    get modelRef(): Reference<CustomEditorModel> {
+        return this._modelRef;
+    }
+    set modelRef(modelRef: Reference<CustomEditorModel>) {
+        this._modelRef = modelRef;
+        this.doUpdateContent();
+        Saveable.apply(this);
+    }
+    get saveable(): Saveable {
+        return this._modelRef?.object;
```

Comment:
If `this._modelRef` can be undefined we should reflect it as such (else we can drop `?`):

```ts
 protected _modelRef: Reference<CustomEditorModel> | undefined;
```

Reply:
fixed